pyam/core.py

```diff
@@ -1361,11 +1378,31 @@ def _meta_idx(data):
     return data[META_IDX].drop_duplicates().set_index(META_IDX).index
 
 
-def _aggregate(df, by):
+def _aggregate(df, by, aggregator='sum', weightvar_df=None):
     """Aggregate `df` by specified column(s), return indexed `pd.Series`"""
     by = [by] if isstr(by) else by
     cols = [c for c in list(df.columns) if c not in ['value'] + by]
-    return df.groupby(cols).sum()['value']
+    # pick aggregator func (default: sum)
+    if aggregator == 'w.avg' and not weightvar_df.empty:
```

Comment:
it won't work if you pass `w.avg` and empty weightvar_df as expected because it will then silently fall back to `sum`

Reply:
agree, one more reason for splitting the weighted-average into another function

Reply:
OK. Agree that silently switching to method='sum' is bad behaviour.
What's your take on "expected behaviour" in that case (method='w.avg' and no weighting var)?

1. - just report ERROR (no aggregation)
2. - silently switch to avg
3. - report problem (WARN/ERROR) and switch to avg
4. - ...

same behaviour in case of weighting var (string) provided - but there's no data?

Reply:
I would report error in both cases as it seems to be inconsistent input